packages/NodeTypeResolver/src/PerNodeTypeResolver/MethodCallTypeResolver.php

```diff
@@ -72,15 +73,18 @@ public function resolve(Node $methodCallNode): array
         return $this->resolveMethodReflectionReturnTypes($methodCallNode, $methodCallVariableType, $methodCallName);
     }
 
-    private function resolveMethodCallName(MethodCall $methodCallNode): ?string
+    private function resolveMethodCallName(MethodCall $methodCallNode): string
```

Comment:
This whole method doesn't make a lot of sense to me. It looks like it picks rather random "names". E.g. for `$foo->$bar()` it gives `"bar"`, for `$foo->{$bar->baz}` it gives `"baz"`, and with this change, for `$foo->{__CLASS__}` it gives `"__CLASS__"`.

What you probably want is to check for `Identifier` only and return null in all other cases (unless you can resolve a dynamic name, of course).

Reply:
I've made some cleanup here - https://github.com/rectorphp/rector/pull/237/commits/1d435beffeb8f3b16c9a69ee71f9311b2686066d - maybe it would be more clear.

I didn't test these cases yet. It should only match non-dynamic values.

> What you probably want is to check for Identifier only and return null in all other cases.

Could you provide some real code so I could better understand?

Reply:
@TomasVotruba If you can guide me, please: what this method should do to `Rector`s? What should it return?

Reply:
This class detects `{something}->methodCall()` type.

So in code:

```php
$variable = {something}->methodCall()
```

we know `$variable` is of certain type

Like in this test: https://github.com/rectorphp/rector/blob/7f66efa095e41b0ee0681efc8b569bd6114f90bc/packages/NodeTypeResolver/tests/PerNodeTypeResolver/AssignTypeResolver/Source/MethodCall.php.inc#L7

I reliazed it could be simpler and standalone tested. I will probably look on it

Reply:
Okay. I'll try to test and fix everthing!